### PR TITLE
New package - itools (toolkit for connecting iOS devices)

### PR DESCRIPTION
**Pull request by bxxxjxxg**
Motivation: create a package for people who want to connect his/her iOS devices

I love this toolchain. 
It helps everyone cross-compile a package within only 3 files.
It's quite pleasing experience to create a spk here.

### Checklist
- [v] Build rule `all-supported` completed successfully
- [v] Package upgrade completed successfully
- [v] New installation of package completed successfully

Add following packages for cross-compiling:
- ifuse
- libimobiledevice
- libplist
- libusb (updated checksum)
- libusbmuxd
- usbmuxd
source: https://github.com/libimobiledevice

Add new package named itools for packing them:
- DSM will start usbmuxd daemon in the background for communication.

After installation, several useful command-line tools are available
in shell. For example, user can use ideviceinfo to see connected
iOS device's information and use ifuse to mount device folder into
system.

Signed-off-by: BingJing Chang <bxxxjxxg@gmail.com>

### CODE REVIEW

File: cross/ifuse/Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+PKG_NAME = ifuse
+PKG_VERS = 1.1.4
+PKG_EXT = tar.gz
+PKG_DIST_NAME = $(PKG_VERS).$(PKG_EXT)
+PKG_DIST_SITE = https://github.com/bxxxjxxg/$(PKG_NAME)/archive
```
Comment (ymartin59) [inline cross/ifuse/Makefile]: Version 1.1.4 does not officially exists. I propose you use commit hash as version and download from https://github.com/libimobiledevice/ifuse/archive/master.zip

File: cross/ifuse/Makefile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
+
+DEPENDS = cross/libplist cross/libimobiledevice cross/fuse
+
+HOMEPAGE = https://github.com/bxxxjxxg/ifuse/
```
Comment (ymartin59) [inline cross/ifuse/Makefile]: Please replace by https://github.com/libimobiledevice/ifuse

File: cross/ifuse/Makefile [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+COMMENT  = A fuse filesystem implementation to access the contents of iOS devices.
+LICENSE  = LGPL-2.1
+
+GNU_CONFIGURE = 1
+PRE_CONFIGURE_TARGET = myPreConfigure
```
Comment (ymartin59) [inline cross/ifuse/Makefile]: We use new naming convention for targets to ease diagnostic. For this one, rename to `ifuse_pre_configure`

File: cross/ifuse/Makefile [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+INSTALL_TARGET = myInstall
+
+include ../../mk/spksrc.cross-cc.mk
+
+myPreConfigure:
+	$(RUN) ; ./autogen.sh
+
+myInstall:
```
Comment (ymartin59) [inline cross/ifuse/Makefile]: The same way, rename to `ifuse_extra_install`

File: cross/ifuse/digests [hunk 5]
```diff
@@ -0,0 +1 @@
+1.1.4.tar.gz MD5 dfc928db8a8b10eeccabcaec14da8d80
```
Comment (ymartin59) [inline cross/ifuse/digests]: Please use `make digests` target so that file contains also sha256 and sha1

File: cross/libplist/Makefile [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+LICENSE  = LGPL-2.1
+
+GNU_CONFIGURE = 1
+PRE_CONFIGURE_TARGET = myPreConfigure
+INSTALL_TARGET = myInstall
+CONFIGURE_ARGS = --without-cython
+
+include ../../mk/spksrc.cross-cc.mk
+
+myPreConfigure:
```
Comment (ymartin59) [inline cross/libplist/Makefile]: Same remark, rename so that targets are unique

File: cross/libimobiledevice/digests [hunk 7]
```diff
@@ -0,0 +1 @@
+1.2.1.tar.gz MD5 47c837c9c146b460cbca0d96f02d7a48
```
Comment (ymartin59) [inline cross/libimobiledevice/digests]: Use `make digests`

File: spk/itools/Makefile [hunk 8]
```diff
@@ -0,0 +1,28 @@
+SPK_NAME = itools
+SPK_VERS = 1.0
+SPK_REV = 1
+SPK_ICON = src/itools.png
+
+DEPENDS = cross/usbmuxd cross/ifuse
+UNSUPPORTED_ARCHS = $(ARM_ARCHES) $(PPC_ARCHES)
```
Comment (ymartin59) [inline spk/itools/Makefile]: What are reasons why ARM and PPC may not run these command line tools?

File: spk/itools/Makefile [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+RELOAD_UI = n
+DISPLAY_NAME = iTools
+STARTABLE = no
+BETA = 1
+CHANGELOG = First Release
+
+HOMEPAGE   = https://www.libimobiledevice.org
+LICENSE    = LGPL 2.1
+
+INSTALLER_SCRIPT = src/installer.sh
```
Comment (ymartin59) [inline spk/itools/Makefile]: Please use generic installer and service support: https://github.com/SynoCommunity/spksrc/wiki/Service-Support

File: spk/itools/Makefile [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+SSS_SCRIPT       = src/dsm-control.sh
+
+INSTALL_PREFIX = /usr/local/$(SPK_NAME)
```
Comment (ymartin59) [inline spk/itools/Makefile]: This is defaults now. Please discard

File: spk/itools/Makefile [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+DISPLAY_NAME = iTools
+STARTABLE = no
+BETA = 1
+CHANGELOG = First Release
+
+HOMEPAGE   = https://www.libimobiledevice.org
+LICENSE    = LGPL 2.1
+
+INSTALLER_SCRIPT = src/installer.sh
+SSS_SCRIPT       = src/dsm-control.sh
```
Comment (ymartin59) [inline spk/itools/Makefile]: Simply replace by `SERVICE_COMMAND = $${SYNOPKG_PKGDEST}/bin/usbmuxd`
If that deamon generates a pid file, please add option to use variable `PID_FILE`

File: spk/itools/src/dsm-control.sh [hunk 12]
```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
```
Comment (ymartin59) [inline spk/itools/src/dsm-control.sh]: That file can be removed thanks to generic service support

File: spk/itools/Makefile (hunk 11 above)
Comment (ymartin59) [reply spk/itools/Makefile]: Question to answer, is that service only runs properly as `root` or is it expected to work properly as non privileged user?

File: spk/itools/src/installer.sh [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+postinst ()
+{
+    # Link
+    ln -s ${SYNOPKG_PKGDEST} ${INSTALL_DIR}
+
+    # Put tmux in the PATH
+    mkdir -p /usr/local/bin
+    ln -s ${INSTALL_DIR}/bin/idevicebackup /usr/local/bin/idevicebackup
```
Comment (ymartin59) [inline spk/itools/src/installer.sh]: I find all these symbolic links "ugly". Commands are easily accessible from `/usr/local/itools/bin` if needed and this location may be added to PATH. I would only create links for command end-user may really invoke himself.

**Comment by ymartin59** [comment]
Thanks for contributing. Looks like interesting.

Sorry but I cannot accept you fork and create your own tag version/releases from `libimobiledevice` organisation repositories.
I propose you request tags there so that it becomes official release, or maybe you request organisation membership so that you can do it on your own.

File: spk/itools/src/installer.sh (hunk 13 above)
Comment (ymartin59) [reply spk/itools/src/installer.sh]: For link creation, you can inspire from recent PR: https://github.com/SynoCommunity/spksrc/pull/3378/files

File: cross/ifuse/Makefile (hunk 1 above)
Comment (bxxxjxxg) [reply cross/ifuse/Makefile]: Thank you. It's useful. Now I can compile the upstream code up to a  specific commit.

File: spk/itools/Makefile (hunk 8 above)
Comment (bxxxjxxg) [reply spk/itools/Makefile]: It cannot pass "make all-supported" tests. 
I simply marked them as unsupported models.
Someone can solve it in the future.

File: spk/itools/Makefile (hunk 11 above)
Comment (bxxxjxxg) [reply spk/itools/Makefile]: It doesn't generate a pid file; however, SVC_WRITE_PID=y is really useful. 
It should be run as root. It requires USB permission.

**Comment by bxxxjxxg** [comment]
Hi @ymartin59,
Thank for your comments and generic service support. It's quite useful.
Would you please review the PR again? 
I can now compile the master branch code up to a specific commit. Thanks!

**Comment by ymartin59** [comment]
May you please reduce list of symbolic links to `/usr/local/bin/`.
And I made a mistake, you should create links from another variable: `ln -s ${SYNOPKG_PKGDEST}/bin/...`

**Comment by bxxxjxxg** [comment]
No problem! I create the symbolic links by using ${SYNOPKG_PKGDEST} now.

May I know your concern for the number of symbolic links to /usr/local/bin/?
I have dropped the link to usbmuxd because it’s a daemon. A daemon doesn’t propose to be started from user. But other binaries named idevice* are the core tools of this package. Although I need just few commands, then I can mount an iOS device as a folder, user may be interested in other commands for other purposes as well. It’s difficult for me to decide it.

**Comment by bxxxjxxg** [comment]
Hi @ymartin59,
Do you have any feedbacks? Thanks!

File: spk/itools/src/service-setup.sh [hunk 14]
```diff
@@ -0,0 +1,53 @@
+
+SERVICE_COMMAND="${SYNOPKG_PKGDEST}/sbin/usbmuxd"
```
[the remark this answers is not in the document]
Comment (bxxxjxxg) [reply spk/itools/src/service-setup.sh]: done.

File: spk/itools/Makefile [hunk 15]
```diff
@@ -0,0 +1,25 @@
+SPK_NAME = itools
+SPK_VERS = 1.0
+SPK_REV = 1
+SPK_ICON = src/itools.png
+
+DEPENDS = cross/usbmuxd cross/ifuse
+UNSUPPORTED_ARCHS = $(ARM_ARCHES) $(PPC_ARCHES)
+
+MAINTAINER = BingJing
```
[the remark this answers is not in the document]
Comment (bxxxjxxg) [reply spk/itools/Makefile]: done.

**Pull request merged**